src/oci-umount.c

```diff
@@ -374,28 +372,15 @@ static int unmount(char *umount_path, bool submounts_only, const struct mount_in
 
 	/* Unmount submounts only */
 	for (i = 0; i < table_sz; i++) {
-		if (!strcmp(umount_path, mnt_table[i].destination)) {
-			found_mnt = true;
-			mntid = mnt_table[i].mntid;
-			break;
+		if (!strncmp(umount_path, mnt_table[i].destination, strlen(umount_path))) {
+			ret = umount2(mnt_table[i].destination, MNT_DETACH);
+			if (!ret)
+				pr_pinfo("Unmounted submount: [%s]\n", mnt_table[i].destination);
+			else
+				pr_perror("Failed to unmount submount: [%s]. Skipping.\n", mnt_table[i].destination);
```

Comment:
Problem with this approach is that what if a parent mount is unmounted first (and hence unmounting  child mount) and later we go to unmount child mount. In that case umount2() will return an error.

I am working on another fix, where if path is not a mount point, we travel up the path and find parent mount and parent mount id and use that instead. I am cleaning up and testing my patches and will create a PR soon.